source/exe/main.cc

```diff
@@ -33,10 +35,32 @@ class ProdComponentFactory : public ComponentFactory {
 
 } // Server
 
+int validateConfigAndExit(OptionsImpl& options, Server::ProdComponentFactory& component_factory,
+                          const LocalInfo::LocalInfoImpl& local_info) {
+  Server::ValidationHotRestart restarter;
+  Logger::Registry::initialize(options.logLevel(), restarter.logLock());
+  Stats::HeapRawStatDataAllocator alloc;
+  Stats::ThreadLocalStoreImpl stats_store(alloc);
+
+  Server::ValidationInstance server(options, restarter, stats_store, restarter.accessLogLock(),
+                                    component_factory, local_info);
+  std::cerr << "configuration '" << options.configPath() << "' OK" << std::endl;
+  server.shutdown();
+  return 0;
+}
+
 int main(int argc, char** argv) {
-  ares_library_init(ARES_LIB_INIT_ALL);
   Event::Libevent::Global::initialize();
   OptionsImpl options(argc, argv, Server::SharedMemory::version(), spdlog::level::warn);
+  Server::ProdComponentFactory component_factory;
+  LocalInfo::LocalInfoImpl local_info(Network::Utility::getLocalAddress(), options.serviceZone(),
+                                      options.serviceClusterName(), options.serviceNodeName());
+
+  if (options.mode() != Server::Mode::Serve) {
+    return validateConfigAndExit(options, component_factory, local_info);
```

Comment:
I think it would be clearer to call it just `validateConfig`, as the `AndExit` implies to me that it's going to do a `::exit(something);`, which it doesn't, it returns.